pkg/webhook/mutation/pod/handler/injection/handle.go

```diff
@@ -77,13 +77,22 @@ func (h *Handler) Handle(mutationRequest *dtwebhook.MutationRequest) error {
 		}
 
 		if !mutated {
-			setNotInjectedAnnotations(mutationRequest, NoMutationNeededReason)
+			annotations.SetNotInjected(
```

Comment:
found a small hiccup in this file, while testing

the code
```go
	if !h.isInputSecretPresent(mutationRequest, bootstrapperconfig.GetSourceConfigSecretName(mutationRequest.DynaKube.Name), consts.BootstrapperInitSecretName) {
		return nil
	}
```

will always run, even if only otelp is configured

We should have a check similar to the one in `otepl/handle.go`, but for the oa/meta injection
```go
func (h *Handler) Handle(mutationRequest *dtwebhook.MutationRequest) error {
	if !mutationRequest.DynaKube.OneAgent().IsAppInjectionNeeded() && !mutationRequest.DynaKube.MetadataEnrichment().IsEnabled(){ ... }
```

Reply:
added with [3141841](https://github.com/Dynatrace/dynatrace-operator/pull/5491/commits/3141841e5261b0cd69f02854eed2bf4aa3428658)